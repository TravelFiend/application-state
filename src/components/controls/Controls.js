import React from 'react';
import PropTypes from 'prop-types';
import styles from './Controls.css';

const Controls = ({ actions, handleSelection }) => (
  <section className={styles.Controls}>
    {actions.map((action) => (
      <button key={action.name} onClick={() => handleSelection(action)}>
        {action.name}
      </button>
    ))}
  </section>
);

Controls.propTypes = {
  actions: PropTypes.arrayOf(PropTypes.func.isRequired).isRequired,
  handleSelection: PropTypes.func.isRequired
};

export default Controls;
