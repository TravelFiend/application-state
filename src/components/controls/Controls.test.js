import React from 'react';
import { shallow } from 'enzyme';
import Controls from './Controls';
import {
  Drink_Coffee,
  Eat_Snack,
  Take_Nap,
  Study
} from '../../actions/incrementors';
 

describe('Controls component', () => {
  let wrapper;
  let handleSelection;
  beforeEach(() => {
    const actions = [Drink_Coffee, Eat_Snack, Take_Nap, Study];
    handleSelection = jest.fn();

    wrapper = shallow(<Controls
      actions={actions}
      handleSelection={handleSelection} />);
  });

  it('renders Controls', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('handlesSelection with the action name', () => {
    wrapper.findWhere(el => el.key() === 'Drink_Coffee').simulate('click');
    expect(handleSelection).toHaveBeenCalled();
  });
});
  
