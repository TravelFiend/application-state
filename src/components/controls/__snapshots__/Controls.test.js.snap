// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls component renders Controls 1`] = `
<section
  className="Controls"
>
  <button
    key="Drink_Coffee"
    onClick={[Function]}
  >
    Drink_Coffee
  </button>
  <button
    key="Eat_Snack"
    onClick={[Function]}
  >
    Eat_Snack
  </button>
  <button
    key="Take_Nap"
    onClick={[Function]}
  >
    Take_Nap
  </button>
  <button
    key="Study"
    onClick={[Function]}
  >
    Study
  </button>
</section>
`;
